libraries/SuplaDevice/src/supla/sensor/MAX6675_K.cpp

```diff
@@ -53,15 +71,16 @@ void MAX6675_K::onInit() {
   channel.setNewValue(getValue());
 }
 
-byte MAX6675_K::spiRead() {
+uint32_t MAX6675_K::spiRead() {
   int i;
-  byte d = 0;
+  uint32_t d = 0;
 
-  for (i = 7; i >= 0; i--) {
+  for (i = 31; i >= 0; i--) {
     digitalWrite(pin_CLK, LOW);
     delay(1);
+    d <<= 1;
     if (digitalRead(pin_DO)) {
-      d |= (1 << i);
+      d |= 1;
     }
```

Comment:
you can even shorten it to: d |= digitalRead(pin_D0);

Reply:
done.
the idea is to keep the name to avoid breaking changes in the code
but you decide if you want to change to "MAXThermocouple"

Reply:
Please rename it.
Anyway we are still doing those changes on "develop" branch, so maintaining of compatibility is limitted. I plan to make an official release of a new SuplaDevice in 1-2 months, so starting from then we'll try to keep compatiblity

Reply:
done

Reply:
thanks!